### ISSUE TITLE
[In App Messaging] FirebaseInAppMessagingDisplay doesn't notify image download failure

### ISSUE THREAD

**Issue by Ethan1983** [firebase/firebase-android-sdk#5002]
### Environment

  * Android Studio version: Android Studio Flamingo | 2022.2.1 Patch 1
  * Firebase Component: In App Messaging - Display & In App Messaging
  * Component version: 20.3.2

### Problem

  `InAppMessagingErrorReason#IMAGE_FETCH_ERROR` isn't notified when `FirebaseInAppMessagingDisplay` fails to download the image.

#### Steps to reproduce:

- Image download could fail for various reasons like network policy etc.
- In such cases, the default display component `FirebaseInAppMessagingDisplay` silently ignores these failures and in app message is ignored.
- Application registered listener via `addDisplayErrorListener` isn't notified of `InAppMessagingErrorReason#IMAGE_FETCH_ERROR`
  

#### Relevant Code:

```
         @Override
          public void onError(Exception e) {
            Logging.loge("Image download failure ");
            if (layoutListener != null) {
              bindingWrapper
                  .getImageView()
                  .getViewTreeObserver()
                  .removeGlobalOnLayoutListener(layoutListener);
            }
            cancelTimers(); // Not strictly necessary.
            inAppMessage = null;
            callbacks = null;
          }
```

**Comment by ione0213** [comment]
My app wasn't notified by FirebaseInAppMessaging after registering the listener via `addDisplayErrorListener`, either.
Is there any update about this issue?

**Comment by lehcar09** [comment]
Hi @Ethan1983, apologies for the late response on the issue. Our engineers have address the issue and released the version [In-App Messaging version 20.4.1](https://firebase.google.com/support/release-notes/android#inappmessaging-ktx_v20-4-1) included in BOM version 33.0.0 for the fix. 

With that, I'll be closing this issue now. Let me know if there's any misunderstanding and/or we need to reopen the issue for further investigation. Thanks!
